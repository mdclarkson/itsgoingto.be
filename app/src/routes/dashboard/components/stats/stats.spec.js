import React from 'react'
import { render } from '@testing-library/react'
import Stats from './stats'

const defaultProps = {
  fetchStats: jest.fn(),
}

describe('(Route) dashoard', () => {
  describe('(Component) stats', () => {
    afterEach(() => {
      jest.clearAllMocks()
    })

    describe('(Lifecycle) componentDidMount', () => {
      it('should call fetchStats', () => {
        render(<Stats {...defaultProps} />)

        expect(defaultProps.fetchStats).toHaveBeenCalledTimes(1)
      })
    })

    describe('(Render)', () => {
      describe('with no stats', () => {
        it('matches snapshot', () => {
          const { asFragment } = render(<Stats {...defaultProps} />)

          expect(asFragment()).toMatchSnapshot()
        })
      })

      describe('with stats', () => {
        it('matches snapshot', () => {
          const { asFragment } = render(<Stats {...defaultProps} polls={15} responses={39} />)

          expect(asFragment()).toMatchSnapshot()
        })
      })
    })
  })
})
