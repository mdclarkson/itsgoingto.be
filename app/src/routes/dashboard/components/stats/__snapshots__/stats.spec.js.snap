// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) dashoard (Component) stats (Render) with no stats matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="info"
  >
    <div
      class="info-block-column"
    >
      <div
        class="info-block"
      >
        <div
          class="info-header"
        >
          Total Polls
        </div>
        <div
          class="info-body"
        >
          -
        </div>
      </div>
    </div>
    <div
      class="info-block-column"
    >
      <div
        class="info-block"
      >
        <div
          class="info-header"
        >
          Total Responses
        </div>
        <div
          class="info-body"
        >
          -
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Route) dashoard (Component) stats (Render) with stats matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="info"
  >
    <div
      class="info-block-column"
    >
      <div
        class="info-block"
      >
        <div
          class="info-header"
        >
          Total Polls
        </div>
        <div
          class="info-body"
        >
          15
        </div>
      </div>
    </div>
    <div
      class="info-block-column"
    >
      <div
        class="info-block"
      >
        <div
          class="info-header"
        >
          Total Responses
        </div>
        <div
          class="info-body"
        >
          39
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
